src/sentry/web/frontend/accept_project_transfer.py

```diff
@@ -0,0 +1,110 @@
+from __future__ import absolute_import
+
+from django import forms
+from django.contrib import messages
+from django.utils.translation import ugettext_lazy as _
+from sentry import roles
+from sentry.web.frontend.base import BaseView
+from django.core.urlresolvers import reverse
+from django.http import HttpResponseRedirect, Http404
+from django.utils.encoding import force_str
+from django.core.signing import BadSignature, SignatureExpired
+from sentry.utils.signing import unsign
+from sentry.models import AuditLogEntryEvent, OrganizationMember, Organization, Team, Project
+
+
+class AcceptProjectTransferForm(forms.Form):
+    team = forms.ChoiceField(choices=[])
+
+    def __init__(self, request, *args, **kwargs):
+        super(AcceptProjectTransferForm, self).__init__(*args, **kwargs)
+        teams = []
+        for o in OrganizationMember.objects.filter(user__email=request.user):
+            org_name = Organization.objects.get(id=o.organization_id).name
+            for t in o.get_teams():
+                option = " %s - %s" % (t.name, org_name)
+                teams.append([t.id, option])
+
+        self.fields['team'].choices = teams
+        self.fields['team'].widget.choices = self.fields['team'].choices
+
+
+class AcceptProjectTransferView(BaseView):
+    required_scope = 'org:admin'
+    sudo_required = True
+
+    def get_form(self, request):
+        if request.method == 'POST':
+            return AcceptProjectTransferForm(request, request.POST, initial=request.POST)
+        return AcceptProjectTransferForm(request)
+
+    def handle(self, request, *args, **kwargs):
+        try:
+            d = request.GET['data']
+        except KeyError:
+            raise Http404
+
+        try:
+            data = unsign(force_str(d))
+        except BadSignature:
+            messages.add_message(
+                request, messages.ERROR,
+                _(u'Could not approve transfer, please make sure link is valid.')
+            )
+            return HttpResponseRedirect(
+                reverse('sentry')
+            )
+
+        try:
+            data = unsign(force_str(d))
```

Comment:
Why are you trying to `unsign` this twice?

I think what you're tyring to do is handle multiple different exceptions. You can do this like:

```python
try:
    data = unsign(force_str(d))
except BadSignature:
    messages.add_message(
        request, messages.ERROR,
        _(u'Could not approve transfer, please make sure link is valid.')
    )
    return HttpResponseRedirect(
        reverse('sentry')
    )
except SignatureExpired:
    messages.add_message(
        request, messages.ERROR,
        _(u'Project transfer link has expired!')
    )
    return HttpResponseRedirect(
        reverse('sentry')
    )
```

You can have multiple `except` clauses.